crates/forge/test_utils/src/runner.rs

```diff
@@ -334,6 +335,14 @@ pub fn assert_builtin(
     builtin: BuiltinName,
     expected_count: usize,
 ) {
+    let scarb_version = ScarbCommand::version().run().unwrap();
+    let expected_count =
+        if builtin == BuiltinName::range_check && scarb_version.scarb >= Version::new(2, 9, 2) {
+            expected_count - 1
+        } else {
+            expected_count
+        };
```

Comment:
I'd add a todo here and create a task to backlog to investigate these differences.

Reply:
Agree, we should tend to minimize such version-related logic.

Reply:
Everyone agrees that such logic is terrible and should be avoided, but it's more of question whether these tests make sense in their current form, given that the values they assert can fluctuate. Added #2806